wptool/wptool.cs

```diff
@@ -197,28 +197,43 @@ class wptool
-						Console.WriteLine("{");
-						Console.WriteLine("\t\"success\": true,");
-						Console.WriteLine("\t\"ip\": \"" + address.ToString() + "\",");
-						Console.WriteLine("\t\"port\": " + destinationPort.ToString() + ",");
-						Console.WriteLine("\t\"osVersion\": \"" + version.ToString() + "\",");
-						Console.WriteLine("\t\"availablePhysical\": " + systemInfo.AvailPhys.ToString() + ",");
-						Console.WriteLine("\t\"totalPhysical\": " + systemInfo.TotalPhys.ToString() + ",");
-						Console.WriteLine("\t\"availableVirtual\": " + systemInfo.AvailVirtual.ToString() + ",");
-						Console.WriteLine("\t\"totalVirtual\": " + systemInfo.TotalVirtual.ToString() + ",");
-						Console.WriteLine("\t\"architecture\": \"" + systemInfo.ProcessorArchitecture.ToString() + "\",");
-						Console.WriteLine("\t\"instructionSet\": \"" + systemInfo.InstructionSet.ToString() + "\",");
-						Console.WriteLine("\t\"processorCount\": " + systemInfo.NumberOfProcessors.ToString() + "");
-						Console.WriteLine("}");
-					}
+                            Version version = new Version(systemInfo.OSMajor, systemInfo.OSMinor, systemInfo.OSBuildNo);
+                            Console.WriteLine("{");
+                            Console.WriteLine("\t\"success\": true,");
+                            Console.WriteLine("\t\"ip\": \"" + address.ToString() + "\",");
+                            Console.WriteLine("\t\"port\": " + destinationPort.ToString() + ",");
+                            Console.WriteLine("\t\"osVersion\": \"" + version.ToString() + "\",");
+                            Console.WriteLine("\t\"availablePhysical\": " + systemInfo.AvailPhys.ToString() + ",");
+                            Console.WriteLine("\t\"totalPhysical\": " + systemInfo.TotalPhys.ToString() + ",");
+                            Console.WriteLine("\t\"availableVirtual\": " + systemInfo.AvailVirtual.ToString() + ",");
+                            Console.WriteLine("\t\"totalVirtual\": " + systemInfo.TotalVirtual.ToString() + ",");
+                            Console.WriteLine("\t\"architecture\": \"" + systemInfo.ProcessorArchitecture.ToString() + "\",");
+                            Console.WriteLine("\t\"instructionSet\": \"" + systemInfo.InstructionSet.ToString() + "\",");
+                            Console.WriteLine("\t\"processorCount\": " + systemInfo.NumberOfProcessors.ToString() + "");
+                            Console.WriteLine("}");
+                        } catch (Exception ex) {
+                            //
+                            // From some point of Windows 10 SDK, IDevice.GetEndPoints throws Exception for Windows Phone 8.1 device.
+                            // Returning ok for now because we can still connect to the device.
+                            //
+                            if (command == "connect" && connectableDevice.Version.Major == 8 && !connectableDevice.IsEmulator()) {
```

Comment:
Note: Interestingly I've seen `ConnectableDevice.Version` returns `8.1` with the updated SDK. This may not be quite expected before as we have been seeing it returns `6.3` for Windows Phone 8.1 device.